cli/module_loader.rs

```diff
@@ -75,6 +75,9 @@ pub async fn load_top_level_deps(factory: &CliFactory) -> Result<(), AnyError> {
   let npm_resolver = factory.npm_resolver().await?;
   if let Some(npm_resolver) = npm_resolver.as_managed() {
     npm_resolver.ensure_top_level_package_json_install().await?;
+    // TODO(nathanwhit): we call `cache_packages` if the lockfile is modified,
+    // so by calling it here it's possible we end up calling it twice
+    npm_resolver.cache_packages().await?;
```

Comment:
It should be doing this within `ensure_top_level_package_json_install`. Does it not happen?

Reply:
Ah, I see:
```
if reqs
      .iter()
      .all(|req| self.resolution.resolve_pkg_id_from_pkg_req(req).is_ok())
    {
      log::debug!(
        "All package.json deps resolvable. Skipping top level install."
      );
      return Ok(()); // everything is already resolvable
    }
```